srtcore/core.cpp

```diff
@@ -7856,7 +7825,32 @@ int srt::CUDT::sendCtrlAck(CPacket& ctrlpkt, int size)
     // There are new received packets to acknowledge, update related information.
     if (CSeqNo::seqcmp(ack, m_iRcvLastAck) > 0)
     {
-        ackDataUpTo(ack);
+        // Sanity check if the "selected ACK" points to a sequence
+        // in the past for the buffer. This SHOULD NEVER HAPPEN because
+        // on drop the loss records should have been removed, and the last received
+        // sequence also can't be in the past towards the buffer.
+
+        // NOTE: This problem has been observed when the packet sequence
+        // was incorrectly removed from the receiver loss list. This should
+        // then stay here as a condition in order to detect this problem,
+        // should it happen in the future.
+        if (CSeqNo::seqcmp(ack, m_pRcvBuffer->getStartSeqNo()) < 0)
+        {
+            // Log and DO NOT UPDATE. Count on that next time
+            // the found out ack will be better.
+
+            LOGC(xtlog.Error,
+                    log << CONID() << "ack: IPE: invalid ACK from %" << m_iRcvLastAck << " to %" << ack << " ("
+                    << CSeqNo::seqoff(m_iRcvLastAck, ack) << " packets) buffer=%" << m_pRcvBuffer->getStartSeqNo());
+        }
+        else
+        {
+            HLOGC(xtlog.Debug,
+                    log << CONID() << "ack: %" << m_iRcvLastAck << " -> %" << ack << " ("
+                    << CSeqNo::seqoff(m_iRcvLastAck, ack) << " packets)");
+
+            m_iRcvLastAck = ack;
```

Comment:
Shouldn't the `m_iRcvLastAck = ack;` happen in any case, even if `ack` is behind the RCV buffer start position?

Reply:
Maybe not. This check was added from https://github.com/Haivision/srt/commit/d84ac20da051e5dd99b0e6a610047f230bad1b58#diff-44bac6e930fb85d14c98d6c23a3d000b7d80d0fe03d374401a9c946f42b01eb9R7731
I think it's better to keep ACK monotonic.

Reply:
The `ack` value is ahead of the `m_iRcvLastAck`. 
A lite ACK could have been sent before with this `ack` position preceding the RCV buffer start position.
What could be the reason to keep the `m_iRcvLastAck` unchanged?

Reply:
Hmm, now that `m_iRcvLastSkipAck` was removed, maybe "m_iRcvLastAck = ack; happen in any case" is better,

Reply:
Ok, I might have missed something, but this condition definitely should be checked and reported. Might be that the position should be shifted forward anyway, even if it would be effectively in the past for the buffer.